src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -540,6 +557,64 @@ public bool Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnl
 				}
 				break;
 
+			case IntrinsicId.Nullable_GetUnderlyingType:
+				foreach (var singlevalue in argumentValues[0].AsEnumerable ()) {
+					AddReturnValue (singlevalue switch {
+						SystemTypeValue systemType => systemType,
+						NullableSystemTypeValue nullableSystemType => new SystemTypeValue (nullableSystemType.UnderlyingType),
+						NullableValueWithDynamicallyAccessedMembers nullableDamValue => nullableDamValue.UnderlyingTypeValue,
+						ValueWithDynamicallyAccessedMembers damValue => damValue,
+						_ => GetMethodReturnValue (calledMethod, returnValueDynamicallyAccessedMemberTypes)
+					});
+				}
+				break;
+
+			case IntrinsicId.Type_MakeGenericType:
+				// Contains part of the analysis done in the Linker, but is not complete
+				// instanceValue here is like argumentValues[0] in linker
+				foreach (var value in instanceValue) {
+					// Nullables without a type argument are considered SystemTypeValues
+					if (!(value is SystemTypeValue typeValue && typeValue.RepresentedType.IsTypeOf ("System", "Nullable`1"))) {
+						// We still don't want to lose track of the type if it is not Nullable<T>
+						AddReturnValue (value);
+						continue;
+					}
+					// We have a nullable
+					foreach (var argumentValue in argumentValues[0]) {
+						if ((argumentValue as ArrayValue)?.TryGetValueByIndex (0, out var underlyingMultiValue) != true) {
+							continue;
+						}
+						foreach (var underlyingValue in underlyingMultiValue) {
+							switch (underlyingValue) {
+							case SystemTypeValue maybeArrayValue when maybeArrayValue.RepresentedType.IsTypeOf ("System", "Array"):
+								AddReturnValue (MultiValueLattice.Top);
+								break;
+							case SystemTypeValue systemTypeValue:
+								AddReturnValue (new NullableSystemTypeValue (typeValue.RepresentedType, systemTypeValue.RepresentedType));
+								break;
+							// Don't warn on Nullable<T> - it will throw instead
+							case NullableValueWithDynamicallyAccessedMembers:
+								AddReturnValue (MultiValueLattice.Top);
+								break;
+							// Generic Parameters and method parameters with annotations
+							case ValueWithDynamicallyAccessedMembers damValue:
+								AddReturnValue (new NullableValueWithDynamicallyAccessedMembers (typeValue.RepresentedType, damValue));
+								break;
+							// Don't warn on Nullable<T>, it will throw instead
+							case NullableSystemTypeValue:
```

Comment:
nit: maybe put this and NullableValueWithDynamicallyAccessedMembers next to each other and have one fall through - since they aren't tracked for the same reason.